sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,1142 @@
+# coding: utf-8
+"""
+Large Margin Nearest Neighbor Classification
+"""
+
+# Author: John Chiotellis <johnyc.code@gmail.com>
+# License: BSD 3 clause
+
+from __future__ import print_function
+from warnings import warn
+
+import sys
+import time
+import numpy as np
+from scipy.optimize import minimize
+from scipy.sparse import csr_matrix, csc_matrix, coo_matrix
+
+from ..base import BaseEstimator, TransformerMixin
+from ..neighbors import NearestNeighbors
+from ..decomposition import PCA
+from ..utils import gen_batches
+from ..utils.extmath import row_norms, safe_sparse_dot
+from ..utils.extmath import _euclidean_distances_without_checks
+from ..utils.random import check_random_state
+from ..utils.multiclass import check_classification_targets
+from ..utils.validation import (check_is_fitted, check_array, check_X_y,
+                                check_scalar)
+from ..externals.six import integer_types, string_types
+from ..exceptions import ConvergenceWarning
+
+
+class LargeMarginNearestNeighbor(BaseEstimator, TransformerMixin):
+    """Distance metric learning for large margin classification.
+
+    Read more in the :ref:`User Guide <lmnn>`.
+
+    Parameters
+    ----------
+    n_neighbors : int, optional (default=3)
+        Number of neighbors to use as target neighbors for each sample.
+
+    n_components : int, optional (default=None)
+        Preferred dimensionality of the embedding.
+        If None it is inferred from ``init``.
+
+    init : string or numpy array, optional (default='pca')
+        Initialization of the linear transformation. Possible options are
+        'pca', 'identity' and a numpy array of shape (n_features_a,
```

Comment:
If init by different options is possible, we may want to alert the user to how this choice/parameter impacts convergence...

Reply:
Sure.